packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -321,18 +290,64 @@ class ProxyTarget {
         // Lookup the schema:
         const fieldKind = this.lookupFieldKind(key);
         // Make the childTargets:
-        const childTargets = mapCursorField(this.cursor, brand(key), (c) => new ProxyTarget(this.context, c));
-        return proxifyField(fieldKind, childTargets);
+        const childTargets = mapCursorField(this.cursor, brand(key), (c) => this.context.createTarget(c));
+        return proxifyField(this.context, fieldKind, childTargets);
     }
 
     /**
-     * @returns the type name of a node.
-     * It shall never be possible to call this for sequence fields as they are unwrapped into arrays.
+     * Sets value of a non-sequence field.
+     * This is correct only if sequence fields are unwrapped into arrays.
      */
-    public getTypeName(key: string): TreeSchemaIdentifier {
-        const childTypes = mapCursorField(this.cursor, brand(key), (c) => c.type);
-        assert(childTypes.length <= 1, 0x3c6 /* invalid non sequence */);
-        return childTypes[0];
+    public setValue(key: string, _value: unknown, typeName: TreeSchemaIdentifier): boolean {
+        const type = { name: typeName, ...this.context.forest.schema.lookupTreeSchema(typeName) };
+        assert(isPrimitive(type), "Cannot set value of a non-primitive field");
+        const target = mapCursorField(this.cursor, brand(key), (c) => this.context.createTarget(c))[0];
+        const path = this.context.forest.anchors.locate(target.prepareAnchorForEdit());
+        assert(path !== undefined, "Cannot locate a path to set a value");
+        return this.context.setNodeValue(path, _value);
+    }
+
+    public insertNode(key: string, cursor: ITreeCursor): boolean {
+        const fieldSchema = getFieldSchema(this.getType() as TreeSchema, key);
+        const fieldKind = getFieldKind(fieldSchema);
+        assert(fieldKind.multiplicity !== Multiplicity.Forbidden, "");
+        // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
+        assert(fieldSchema.types !== undefined && fieldSchema.types?.has(cursor.type), "Unknown type");
+        const path = this.context.forest.anchors.locate(this.prepareAnchorForEdit());
+        assert(path !== undefined, "Cannot locate a path to insert a node");
+        return this.context.insertNode({
+            parent: path,
+            parentField: brand(key),
+            parentIndex: 0,
```

Comment:
This will always insert at the start of the field.
If you're making the assumption that the field is empty then it would be good to document and validate the assumption.
If not, I'm curious why we're not just taking an index as input and passing it here.

Reply:
Added a validation and some documentation for the method.